### PR TITLE
Join and Leave Projects from Project Embeds

### PR DESCRIPTION
**Pull request by ttseng**
Created a new remix because a bunch of the old commits weren't appearing after our reverse.

This is a continuation of Tara's work on adding a join / leave button to project pages that was accidentally merged a week ago 

## Links
* Remix: https://colorful-aries.glitch.me/~pupdates-editor
* Issue-tracker: https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/glitch-community-work/503

## Changes:
* There is a new button below Project Embeds that lets you join or leave a project.  Project Embeds appear in these places: project pages, user profiles, collections, and team pages.
* When you leave or join a project, the project users listed in the Profile List Container should update accordingly (you should see your avatar be added and removed)
* The buttons below are responsive and change text on smaller screens so they can all fit.

## GIF/Screenshots:
refresh-embed
Joining and Leaving Projects from a project page project embed

image
The button text changes for small viewports

## How To Test:
* log into the remix
* go to a project page like https://colorful-aries.glitch.me/~pupdates-editor and click the join/leave project button.

### CODE REVIEW

File: src/presenters/pages/project.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import React, { useState, useEffect } from 'react';
 import { withRouter } from 'react-router-dom';
 import PropTypes from 'prop-types';
+import { Helmet } from 'react-helmet-async';
 import { Button, Icon, Loader } from '@fogcreek/shared-components';
 
 import Heading from 'Components/text/heading';
 import Markdown from 'Components/text/markdown';
 import NotFound from 'Components/errors/not-found';
-import GlitchHelmet from 'Components/glitch-helmet';
```
Comment (whimsicallyson) [inline src/presenters/pages/project.js]: I think we want to keep this, not go back to the React helmet

File: src/presenters/pages/project.js [hunk 2]
```diff
@@ -27,14 +27,13 @@ import { useCurrentUser } from 'State/current-user';
 import { useToggleBookmark } from 'State/collection';
 import { useProjectEditor, useProjectMembers } from 'State/project';
 import { getUserLink } from 'Models/user';
-import { userIsProjectMember, userIsProjectAdmin, getProjectLink, getProjectAvatarUrl, SUSPENDED_AVATAR_URL } from 'Models/project';
+import { userIsProjectMember, userIsProjectAdmin } from 'Models/project';
+import { addBreadcrumb } from 'Utils/sentry';
 import { getAllPages } from 'Shared/api';
 import useFocusFirst from 'Hooks/use-focus-first';
+import useDevToggle from 'State/dev-toggles';
```
Comment (whimsicallyson) [inline src/presenters/pages/project.js]: My stuff launched! we don't need this dev toggle anymore

File: src/presenters/pages/project.js [hunk 3]
```diff
@@ -64,7 +65,7 @@ const ReadmeError = (error) =>
       This project would be even better with a <code>README.md</code>
     </>
   ) : (
-    <>We couldn't load the readme. Try refreshing?</>
+    <>We couldn{"'"}t load the readme. Try refreshing?</>
```
Comment (whimsicallyson) [inline src/presenters/pages/project.js]: is this how we're supposed to do quotes? I've never seen it before, and it doesn't feel intuitive...

File: src/components/project/leave-project-pop.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import { useTrackedFunc } from 'State/segment-analytics';
+
+const LeaveProjectPopover = ({ project, leaveProject, togglePopover, align }) => {
+  const trackLeaveProject = useTrackedFunc(leaveProject, 'Leave Project clicked');
+
+  return (
+    <PopoverDialog wide focusOnDialog align={align}>
+      <PopoverTitle>Leave {project.domain}</PopoverTitle>
+      <PopoverActions>
+        <Icon icon="wave" />
```
Comment (whimsicallyson) [inline src/components/project/leave-project-pop.js]: Screen Shot 2019-10-17 at 3 49 23 PM
such a tiny hand!

File: src/presenters/pages/project.js [hunk 5]
```diff
@@ -161,28 +163,9 @@ const ProjectPage = ({ project: initialProject }) => {
...
-      'A simple Node app with a SQLite database to hold app data.',
-    ]);
-    const defaultProjectDescriptionPattern = /(A|The) [a-z]{2,} project that does [a-z]{2,} things/g;
-    const usesDefaultDescription = helloTemplateDescriptions.has(project.description) || defaultProjectDescriptionPattern.test(project.description);
-    if (!project.description || usesDefaultDescription || project.suspendedReason) {
-      return `Check out ~${project.domain} on Glitch, the ${tagline}`;
-    }
-    return `${renderText(project.description)} 🎏 Glitch is the ${tagline}`;
-  }, [project.domain, project.description, project.suspendedReason, tagline]);
-
```
Comment (whimsicallyson) [inline src/presenters/pages/project.js]: why did you get rid of this section?

File: src/presenters/pages/project.js (hunk 3 above)
Comment (keithk) [reply src/presenters/pages/project.js]: I wonder if prettier did this? 👀

**Comment by keithk** [comment]
Wow, pull request #1000! 

[ch1738]

**Comment by keithk** [comment]
trying again (sry i'm using this PR to test) [ch1738]

**Comment by keithk** [comment]
another try 🙃  [ch1738]

**Comment by keithk** [comment]
one more time [ch1738]

File: src/components/project/leave-project-pop.js (hunk 4 above)
Comment (ttseng) [reply src/components/project/leave-project-pop.js]: removed it since it's not actually needed here!

**Comment by ttseng** [comment]
@whimsicallyson just one more thing that needs to be added - I'm adding a way to refresh the embed contents when you join or leave a project, since it's not good that you don't see your avatar added or removed from the embed user list.

**Comment by ttseng** [comment]
refresh-embed

Made changes to `src/components/project/project-embed` to refresh the embed whenever a user joins or leaves a project.

**Pull request merged**